stdlib/src/builtin/string_literal.mojo

```diff
@@ -389,19 +390,35 @@ struct StringLiteral(
 
     @always_inline
     fn as_bytes(ref [_]self) -> Span[UInt8, __origin_of(self)]:
-        """Returns a contiguous slice of the bytes owned by this string.
+        """Returns a contiguous slice of bytes.
 
         Returns:
-            A contiguous slice pointing to the bytes owned by this string.
+            A contiguous slice pointing to bytes.
 
         Notes:
             This does not include the trailing null terminator.
         """
 
-        # Does NOT include the NUL terminator.
         return Span[UInt8, __origin_of(self)](
-            unsafe_ptr=self.unsafe_ptr(),
-            len=self.byte_length(),
+            unsafe_ptr=self.unsafe_ptr(), len=self.byte_length()
+        )
+
+    @always_inline
+    fn as_bytes_read[O: ImmutableOrigin](ref [O]self) -> Span[UInt8, O]:
```

Comment:
I think these can be written as `return self.as_bytes().get_immutable()`.

Reply:
I applied your suggestion to `String`, I didn't find any other way to get the trait to work for `StringLiteral` and `StringSlice` since they both give a "non self related origin" with their "as_bytes()". I'm open to suggestions though since I was just tinkering trying to get everything to behave nicely. 

I've also now removed the "as_bytes(ref [_]self)" from `StringLiteral` because when I removed the method that returns `StaticConstantOrigin` the ref method gives `error: cannot form reference to borrowed register value`

Reply:
Thanks for investigating! I don't know what's best/correct way to use them either. We can get the part where you are sure working well in first, and iterate. WDYT?